core/sugartoir.ml

```diff
@@ -879,7 +880,7 @@ struct
                    let env, bindings =
                      List.fold_left2
                        (fun (env, bindings) (body, p) t ->
-                         let p, penv = CompilePatterns.desugar_pattern `Local p in
+                         let p, penv = CompilePatterns.desugar_pattern Scope.Local p in
```

Comment:
There is no single place in the code that would call `desugar_pattern` with `Scope.Global`. Remove this argument.